pkg/restic/exec_commands.go

```diff
@@ -183,8 +183,8 @@ func getSummaryLine(b []byte) ([]byte, error) {
 
 // RunRestore runs a `restic restore` command and monitors the volume size to
 // provide progress updates to the caller.
-func RunRestore(restoreCmd *Command, log logrus.FieldLogger, updateFunc func(velerov1api.PodVolumeOperationProgress)) (string, string, error) {
-	snapshotSize, err := getSnapshotSize(restoreCmd.RepoIdentifier, restoreCmd.PasswordFile, restoreCmd.CACertFile, restoreCmd.Args[0], restoreCmd.Env)
+func RunRestore(restoreCmd *Command, log logrus.FieldLogger, updateFunc func(velerov1api.PodVolumeOperationProgress), insecureTLS string) (string, string, error) {
```

Comment:
The `restoreCmd` already has the insecure flag right?  so you don't need to pass the param ` insecureTLS string`?

Sorry I didn't notice this in the first pass.

Reply:
This is the tricky part of Restic restore command.
This function calls two Restic commands: `stats` command to get snapshot size, then the `restore` command to do the data restore.
restoreCmd already has the insecureTLS flag in extraFlags parameter, but the `stats` command doesn't.
The `stats` command is called in function `getSnapshotSize`.  Parts of its parameters are copied from `restore` command, but the extraFlags is not. So it's not easy to tell what to do without a new flag.

Reply:
Removed the insecureTLS parameter.
Retrieve from extraFlags array.